api/src/shop/stripe_checkout.py

```diff
@@ -0,0 +1,252 @@
+  resp = stripe.Subscription.list(customer=stripe_customer_id)
+  ret = []
+  for sub in resp["data"]:
+    if (active_only):
+      if (sub['status'] in [
+        SubscriptionStatus.ACTIVE,
+        SubscriptionStatus.TRAILING,
+        ]):
+        ret.append(sub)
+    else:
+      ret.append(sub)
+
+  return ret
+
+
+def get_stripe_customer(member_info:Member):
+  try:
+    customer_search_result = stripe.Customer.search(
+      query=f"metadata['{MSMetaKeys.MEMBER_NUMBER}']:'{member_info.member_number}'"
+      )
+
+    # If we find exactly one customer matching the e-mail, we return it
+    if (len(customer_search_result.data) == 1):
+      customer = customer_search_result.data[0]
+
+      # Update the metadata if needed
+      if (MSMetaKeys.USER_ID not in customer.metadata or
+          MSMetaKeys.MEMBER_NUMBER not in customer.metadata):
+        stripe.Customer.modify(customer["id"],
+          metadata = {
+            MSMetaKeys.USER_ID       : member_info.member_id,
+            MSMetaKeys.MEMBER_NUMBER : member_info.member_number,
+          })
+      return customer
+
+    # If no customer is found, we create one
+    if (len(customer_search_result.data) == 0):
+      customer = stripe.Customer.create(
+        description="Created by Makeradmin",
+        email=member_info.email,
+        name=f"{member_info.firstname} {member_info.lastname}",
+        metadata = {
+          MSMetaKeys.USER_ID : member_info.member_id,
+          MSMetaKeys.MEMBER_NUMBER : member_info.member_number,
+        }
+      )
+      return customer
+
+    # FIXME: Can we have multiple customers with the same e-mail???
+    for customer in customer_search_result:
+      if (MSMetaKeys.MEMBER_NUMBER in customer.metadata and int(customer.metadata[MSMetaKeys.MEMBER_NUMBER]) == member_info.member_id):
+        return customer
+
+  except Exception as e:
+    print(f"Encountered a problem: {e}")
+  
+  return None
+
+# Helper that can look up relevant price for the given member/subscription combo
+# This would be a good place to implement discounts as the member_info could give
+# the user a different price compared to the standard price for the products.
+def lookup_subscription_price_for(member_info:Member, subscription_type:SubscriptionTypes):
+    # FIXME: Really bad to have hardcoded prices...
+    if (subscription_type == SubscriptionTypes.MEMBERSHIP):
+      return 'price_1MPw5kKMK5LCTpGFzgJAUIID'
+    if (subscription_type == SubscriptionTypes.LAB):
+      return 'price_1MPw64KMK5LCTpGFfj1fCWdf'
+    return None
+
+def calc_start_ts(current_end_date:datetime.date) -> int:
+  dt = datetime.datetime.combine(current_end_date, datetime.datetime.min.time())
+  tmp = int(dt.timestamp())
+  utc_now = int(datetime.datetime.utcnow().timestamp())
+
+  # If the trial lasts for more than a day, we reduce the period by a day to 
+  # make sure we start billing the customer in time.
+  if (tmp - utc_now > 86400):
+    return tmp - 86400
+  return 'now'
```

Comment:
Känns vettigare att använda timedeltas istället för att hålla på och mixtra med timestamps.
```suggestion
  dt = datetime.datetime.combine(current_end_date, datetime.time(0, 0))
  utc_now = datetime.datetime.utcnow()

  # If the trial lasts for more than a day, we reduce the period by a day to 
  # make sure we start billing the customer in time.
  if (dt - utc_now > datetime.timedelta(days=1):
    return int((tmp - datetime.timedelta(days=1)).timestamp())
  return 'now'
```